kimchi/src/verifier.rs

```diff
@@ -492,6 +492,12 @@ where
     //~ Essentially, this steps verifies that $f(\zeta) = t(\zeta) * Z_H(\zeta)$.
     //~
 
+    if proof.prev_challenges.len() != index.prev_challenges {
```

Comment:
can you add a spec comment for this step? You can steal it from here: https://github.com/o1-labs/proof-systems/pull/609/files#diff-fc95db4d289fd33bc7fbefe8a487926ce8b47cbe71f047a1dd6933d7444b5d5aR146